corporate/views/remote_billing_page.py

```diff
@@ -185,3 +193,46 @@ def remote_billing_page_server(request: HttpRequest, *, server_uuid: PathOnly[st
 @typed_endpoint
 def remote_billing_page_realm(request: HttpRequest, *, realm_uuid: PathOnly[str]) -> HttpResponse:
     return remote_billing_page_common(request, realm_uuid=realm_uuid, server_uuid=None)
+
+
+@authenticated_remote_realm_management_endpoint
+@typed_endpoint
+def remote_realm_sponsorship(
+    request: HttpRequest,
+    remote_realm: RemoteRealm,
+    *,
+    user: Json[UserDataForRemoteBilling],
+    realm: Json[RealmDataForAnalytics],
+    organization_type: Json[str],
+    website: Json[str],
+    description: Json[str],
+    expected_total_users: Json[str],
+    paid_users_count: Json[str],
+    paid_users_description: Json[str],
+) -> HttpResponse:
+    billing_session = RemoteRealmBillingSession(remote_realm, user=user)
+    post_data = request.POST.copy()
+    form = SponsorshipRequestForm(post_data)
+    billing_session.request_sponsorship(form)
+    return json_success(request)
+
+
+@typed_endpoint
+def remote_server_sponsorship(
+    request: HttpRequest,
+    remote_server: RemoteZulipServer,
+    *,
+    user: Json[UserDataForRemoteBilling],
+    realm: Optional[Json[RealmDataForAnalytics]] = None,
+    organization_type: Json[str],
+    website: Json[str],
+    description: Json[str],
+    expected_total_users: Json[str],
+    paid_users_count: Json[str],
+    paid_users_description: Json[str],
+) -> HttpResponse:
+    billing_session = RemoteServerBillingSession(remote_server, user=user)
+    post_data = request.POST.copy()
+    form = SponsorshipRequestForm(post_data)
+    billing_session.request_sponsorship(form)
+    return json_success(request)
```

Comment:
As noted previously, please don't add functions like this here; we are putting the remote versions of functions just after the one that they are a variant of; this will make it way more maintainable.